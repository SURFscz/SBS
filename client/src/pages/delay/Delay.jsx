import React, {useEffect, useState} from "react";
import "./Delay.scss";
import CountDownDialog from "../../components/countdown-dialog/CountDownDialog";
import Waiting from "../../icons/undraw_season_change_f99v.svg";

const counterTimeOut = 20;

export default function Delay({history}) {

    const [counter, setCounter] = useState(counterTimeOut);
    const [serviceName, setServiceName] = useState(null);

    useEffect(() => {
        const urlSearchParams = new URLSearchParams(window.location.search);
        setServiceName(urlSearchParams.get("service_name"));
    }, [])

    if (counter > 0) {
        setTimeout(() => setCounter(counter - 1), 1000);
    } else {
        history.push(`/service-aup${window.location.search}`);
    }

    return (
        <div className="mod-delay">
            <CountDownDialog
                serviceName={serviceName}
                isOpen={true}
                counter={counter}/>
            <img className="waiting"
                 src={Waiting}
                 alt="waiting"/>
        </div>
    )
}
