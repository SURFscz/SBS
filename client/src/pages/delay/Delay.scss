@import "../../stylesheets/vars";

.mod-delay {
    max-width: $compact-medium;
    margin: 25px auto 40px auto;
    width: 100%;

    @media (max-width: 680px) {
        padding: 0 15px;
    }

    h1 {
        margin: 25px 0;
    }

    img {
        width: 370px;
        display: block;
        margin: 0 auto;
    }

}
