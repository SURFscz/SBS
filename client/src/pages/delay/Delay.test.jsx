import React from "react";
import { render } from "@testing-library/react";
import Delay from "./Delay";

describe("Delay", () => {
    it("matches snapshot with default props", () => {
        const { asFragment } = render(
            <Delay />
        );
        expect(asFragment()).toMatchSnapshot();
    });

    it("matches snapshot with custom title", () => {
        const { asFragment } = render(
            <Delay title="Custom Title" />
        );
        expect(asFragment()).toMatchSnapshot();
    });

    it("matches snapshot with custom description", () => {
        const { asFragment } = render(
            <Delay description="Custom Description" />
        );
        expect(asFragment()).toMatchSnapshot();
    });
});
