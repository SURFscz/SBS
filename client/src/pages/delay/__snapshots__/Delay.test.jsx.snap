// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Delay matches snapshot with custom description 1`] = `
<DocumentFragment>
  <div
    class="mod-delay"
  >
    <div
      class="sds--modal sds--backdrop"
    >
      <div
        class="sds--modal--container  count-down-dialog "
      >
        <div
          class="sds--modal--title "
        >
          <p
            class="sds--text--h4"
          >
            Access to [missing "{{name}}" value]
          </p>
        </div>
        <div
          class="sds--modal--content"
        >
          <div
            class="sds--text--rich"
          >
            <section
              class="count-down-content"
            >
              <p>
                To access application [missing "{{name}}" value] you will have to be provisioned. This unfortunately takes some time, grab a cup of ☕️.
              </p>
              <p>
                You will be redirected in 
                <span
                  class="counter"
                >
                  20
                </span>
                 seconds.
              </p>
            </section>
          </div>
          <div
            class="sds--modal--actions sds--actions--outer sds--space--top--5"
          />
        </div>
      </div>
    </div>
    <img
      alt="waiting"
      class="waiting"
      src="undraw_season_change_f99v.svg"
    />
  </div>
</DocumentFragment>
`;

exports[`Delay matches snapshot with custom title 1`] = `
<DocumentFragment>
  <div
    class="mod-delay"
  >
    <div
      class="sds--modal sds--backdrop"
    >
      <div
        class="sds--modal--container  count-down-dialog "
      >
        <div
          class="sds--modal--title "
        >
          <p
            class="sds--text--h4"
          >
            Access to [missing "{{name}}" value]
          </p>
        </div>
        <div
          class="sds--modal--content"
        >
          <div
            class="sds--text--rich"
          >
            <section
              class="count-down-content"
            >
              <p>
                To access application [missing "{{name}}" value] you will have to be provisioned. This unfortunately takes some time, grab a cup of ☕️.
              </p>
              <p>
                You will be redirected in 
                <span
                  class="counter"
                >
                  20
                </span>
                 seconds.
              </p>
            </section>
          </div>
          <div
            class="sds--modal--actions sds--actions--outer sds--space--top--5"
          />
        </div>
      </div>
    </div>
    <img
      alt="waiting"
      class="waiting"
      src="undraw_season_change_f99v.svg"
    />
  </div>
</DocumentFragment>
`;

exports[`Delay matches snapshot with default props 1`] = `
<DocumentFragment>
  <div
    class="mod-delay"
  >
    <div
      class="sds--modal sds--backdrop"
    >
      <div
        class="sds--modal--container  count-down-dialog "
      >
        <div
          class="sds--modal--title "
        >
          <p
            class="sds--text--h4"
          >
            Access to [missing "{{name}}" value]
          </p>
        </div>
        <div
          class="sds--modal--content"
        >
          <div
            class="sds--text--rich"
          >
            <section
              class="count-down-content"
            >
              <p>
                To access application [missing "{{name}}" value] you will have to be provisioned. This unfortunately takes some time, grab a cup of ☕️.
              </p>
              <p>
                You will be redirected in 
                <span
                  class="counter"
                >
                  20
                </span>
                 seconds.
              </p>
            </section>
          </div>
          <div
            class="sds--modal--actions sds--actions--outer sds--space--top--5"
          />
        </div>
      </div>
    </div>
    <img
      alt="waiting"
      class="waiting"
      src="undraw_season_change_f99v.svg"
    />
  </div>
</DocumentFragment>
`;
